server/src/main/java/org/opensearch/search/aggregations/AggregationCollectorManager.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.aggregations;
+
+import org.apache.lucene.search.Collector;
+import org.apache.lucene.search.CollectorManager;
+import org.opensearch.common.CheckedFunction;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.profile.query.InternalProfileCollector;
+import org.opensearch.search.query.ReduceableSearchResult;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Common {@link CollectorManager} used by both concurrent and non-concurrent aggregation path and also for global and non-global
+ * aggregation operators
+ */
+public class AggregationCollectorManager implements CollectorManager<Collector, ReduceableSearchResult> {
+    private final SearchContext context;
+    private final CheckedFunction<SearchContext, List<Aggregator>, IOException> aggProvider;
+    private final String collectorReason;
+
+    protected Collector collector;
```

Comment:
This `collector` is not used anywhere in this class, please remove it

Reply:
this is used by derived classes

Reply:
So declare it in derived classes please, this class has no use of the collector

Reply:
It is common across all the derived classes and is initialized in constructor such that any issues with the collector setup is caught during construction time.

Reply:
You can do initialization in derived classes, the point is: the collector is no use for base class at all, the derived classes could enhance the base class with anything they need to, but base class could not (and should not) oversee what the derived classes need or may need in the future.

Reply:
Updated this. I was trying to enforce that `collector` instance is initialized for all the derived implementation by keeping it in base class such that any error is caught during setup instead of delaying it in `newCollector`. But I see your point of `collector` not being used in base class and probably it should be left up to derived classes when they want to initialize `collector` instance.